other/lru_cache.py

```diff
@@ -1,49 +1,55 @@
-class Double_Linked_List_Node():
+from typing import Optional, Callable
```

Comment:
Please sort imports so that Callable comes before Optional.  Doing so improves readability and makes it difficult to add duplications.